src/main/scala/mesosphere/marathon/api/akkahttp/v2/PodsController.scala

```diff
@@ -204,9 +202,32 @@ class PodsController(
       }
     }
 
-  def versions(podId: PathId): Route = ???
+  def versions(podId: PathId): Route =
+    authenticated.apply { implicit identity =>
+      podManager.find(podId) match {
+        case None =>
+          reject(Rejections.EntityNotFound.noPod(podId))
+        case Some(pod) =>
+          authorized(ViewRunSpec, pod).apply {
+            val versions = podManager.versions(podId).runWith(Sink.seq)
+            complete(versions)
```

Comment:
`pods.raml` says the return type here is `type: string[]` where `podManager.versions` gives you a `Source[Timestamp...]`. hmm...

Reply:
If all versions are datetimes then I can adapt. I always thought that versions are strings but internally datetimes.